dex/farm/src/farm_token.rs

```diff
@@ -160,23 +158,19 @@ pub trait FarmTokenModule:
 
     fn burn_farm_tokens_from_payments(
         &self,
-        payments: &ManagedVec<EsdtTokenPayment<Self::Api>>,
-    ) -> SCResult<()> {
+        payments: ManagedVecIterator<EsdtTokenPayment<Self::Api>>,
+    ) {
+        let mut total_amount = BigUint::zero();
         for entry in payments {
-            self.burn_farm_tokens(&entry.token_identifier, entry.token_nonce, &entry.amount)?;
+            total_amount += &entry.amount;
+            self.send()
+                .esdt_local_burn(&entry.token_identifier, entry.token_nonce, &entry.amount);
         }
-        Ok(())
+        self.farm_token_supply().update(|x| *x -= total_amount);
     }
 
-    fn burn_farm_tokens(
-        &self,
-        farm_token_id: &TokenIdentifier,
-        farm_token_nonce: Nonce,
-        amount: &BigUint,
-    ) -> SCResult<()> {
-        let farm_amount = self.get_farm_token_supply();
-        require!(&farm_amount >= amount, "Not enough supply");
-        self.nft_burn_tokens(farm_token_id, farm_token_nonce, amount);
-        Ok(())
+    fn burn_farm_tokens(&self, token_id: &TokenIdentifier, nonce: Nonce, amount: &BigUint) {
+        self.send().esdt_local_burn(token_id, nonce, amount);
+        self.farm_token_supply().update(|x| *x -= amount);
```

Comment:
why remove the check that `total_supply > amount_to_burn`

Reply:
to have less SCResults in file. the check was a sanity check and should pass always. if not pass, it will fail anyway since biguints substraction would result in negative. so the outcome is the same, only the error message differs